### PR TITLE
[FL-870] Auto-generated firmware documentation

### PR DESCRIPTION
**Pull request by drunkbatya**
# What's new

- Add generating documentation CI/CD on push to dev branch
- Add uploading documentation CI/CD to Github Pages

# Verification 

- Push something to dev branch, updated documentation will be available on 'https://flipperdevices.github.io/flipperzero-firmware/'

# Checklist (For Reviewer)

- [ ] PR has description of feature/bug or link to Confluence/Jira task
- [ ] Description contains actions to verify feature/bugfix
- [ ] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+env:
+  TARGETS: f7
+  DEFAULT_TARGET: f7
+  FBT_TOOLCHAIN_PATH: /runner/_work
+
+jobs:
+  doxygen:
+    if: ${{ !github.event.pull_request.head.repo.fork }}
+    runs-on: [self-hosted, FlipperZeroShell]
```
Comment (skotopes) [inline .github/workflows/docs.yml]: I don't see reasons why we should run it on our runners.

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+name: 'Generate documentation with Doxygen'
+
+on:
+  push:
+    branches:
+      - dev
```
Comment (skotopes) [inline .github/workflows/docs.yml:7]: We do need tags too:
- Production: all tags separately
- Dev: overwriting

**Review by skotopes — CHANGES_REQUESTED** [review]

File: documentation/Doxyfile.cfg [hunk 3]
```diff
@@ -2021,7 +2035,7 @@ RTF_EXTENSIONS_FILE    =
 
 GENERATE_MAN           = NO
 
-# The MAN_OUTPUT tag is used to specify where the man pages will be put. If a
+# The MAN_OUTPUT tag is used to specify where the man И will be put. If a
```
Comment (giacomoferretti) [inline documentation/Doxyfile.cfg]: Out of curiosity, why this change? I know it's just a comment and that it doesn't impact the docs generation.

Comment (skotopes) [reply documentation/Doxyfile.cfg]: looks like some replace artifact

Comment (Astrrra) [reply documentation/Doxyfile.cfg]: Oops, I think I just accidentally typed that, fixed now

**Comment by Astrrra** [comment]
Superseded by #2944

**Pull request closed without merging**